File: apps/web/src/code/js/basics/process_exit.js
import { Sandbox } from 'e2b'

const sandbox = await Sandbox.create({
  template: 'base',
  onExit: () => console.log('[sandbox]', 'process ended'), // $HighlightLine
})

const proc = await sandbox.process.start({cmd: 'echo "Hello World!"'})
await proc.wait()
// output: [sandbox] process ended

const procWithCustomHandler = await sandbox.process.start({
  cmd: 'echo "Hello World!"',
  onExit: () => console.log('[process]', 'process ended'), // $HighlightLine
})
await procWithCustomHandler.wait()
// output: [process] process ended

await sandbox.close()

File: apps/web/src/code/python/basics/process_exit.py
from e2b import Sandbox

sandbox = Sandbox(
    template="base",
    on_exit=lambda: print("[sandbox]", "process ended"),  # $HighlightLine
)

proc = sandbox.process.start('echo "Hello World!"')
proc.wait()
# output: [sandbox] process ended

proc_with_custom_handler = sandbox.process.start(
    'echo "Hello World!"',
    on_exit=lambda: print("[process]", "process ended"),  # $HighlightLine
)
proc_with_custom_handler.wait()
# output: [process] process ended

sandbox.close()
